core/templates/dev/head/services/NavigationService.js

```diff
@@ -0,0 +1,84 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for navigating the page with tab and shift-tab
+ */
+oppia.factory('NavigationService', [
+  '$rootScope', function($rootScope) {
+    $rootScope.activeMenuName = '';
```

Comment:
I was focused at getting off the issue first, but one thing to note is why declare this variable as `$rootScope`? In general I think `$rootScope` should be declared when a particular variable should be used over multiple files. 

Since here we need `ACTION_OPEN` (and others) for over just two files, we should not declare this as `rootScope`. So we can do something as add these variables to `navigation` variable as you have added the functions. Then in the respective js files do `$scope.ACTION_OPEN = NavigationService.ACTION_OPEN`.

Reply:
I added all the $rootScope variables except activeMenuName to navigation and in NavigationService replaced $rootScope.ACTION_OPEN with navigation.ACTION_OPEN

<img width="1440" alt="screenshot 2018-12-25 at 1 26 35 am" src="https://user-images.githubusercontent.com/35570715/50406060-3682f200-07e4-11e9-8c72-515b69ee437a.png">

I also added $scope.ACTION_OPEN = NavigationService.ACTION_OPEN to the respective JS file but it doesn't seem to work

Reply:
Should I remove $root from the HTML template?

Reply:
Yes, why not?

I think you should try to understand why and when do we need $root and other things, just not do blindly what reviewers say. Please don't rush to get PR's in, instead try to learn how things work. It's okay to ask doubts but ask when you're completely exhausted (after searching and all..). I would be more than happy to help you, whenever you'll need :).

Thanks

Reply:
Sure

Reply:
Why the comment has been deleted? This is seriously not good on your part.

Reply:
Okay, no problem.

Reply:
@bansalnitish , Do you have any suggestions on what to do with activeMenuName. If I remove it from the rootScope then how should I return the value back to the Directive JS file?

Reply:
@bansalnitish Sorry for bothering you but I just figured it out. I will resolve the merge conflicts and push the changes. Thanks :)